cryptoki/src/session/object_management.rs

```diff
@@ -3,66 +3,249 @@
 //! Object management functions
 
 use crate::context::Function;
-use crate::error::{Result, Rv, RvError};
+use crate::error::{Error, Result, Rv, RvError};
 use crate::object::{Attribute, AttributeInfo, AttributeType, ObjectHandle};
 use crate::session::Session;
 use cryptoki_sys::*;
 use std::collections::HashMap;
 use std::convert::TryInto;
+use std::num::NonZeroUsize;
 
 // Search 10 elements at a time
 const MAX_OBJECT_COUNT: usize = 10;
```

Comment:
AFAIK `NonZeroUsize::new` is const and could be directly initialized here:

```suggestion
const MAX_OBJECT_COUNT: NonZeroUsize = NonZeroUsize::new(10).expect("non-zero value to be passed");
```

Reply:
`expect()` is not stable yet to be assigned to a `const`. In this instance, I believe it is OK to use `new_unchecked()` and document the safety condition in the code.